### PR TITLE
Add mini firebase auth SDK

### PR DESCRIPTION
**Pull request by sethvargo**
**Release Note**

```release-note
NONE
```

### CODE REVIEW

File: pkg/internal/firebase/firebase.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
+// Copyright 2020 Google LLC
```
Comment (mikehelmick) [inline pkg/internal/firebase/firebase.go:1]: let's put these in just

/internal

I don't want anyone depending on our repo to provide this.

**Review by whaught — APPROVED** [review]
/lgtm

**Review by mikehelmick — CHANGES_REQUESTED** [review]

File: pkg/internal/firebase/firebase.go (hunk 1 above)
Comment (sethvargo) [reply pkg/internal/firebase/firebase.go:1]: They are 😄

**Review by mikehelmick — APPROVED** [review]

**Pull request merged**